File: Socialite.Domain/Entities/Status.cs
using System;
using Socialite.Domain.Common;
using Socialite.Domain.Interfaces;

namespace Socialite.Domain.Entities
{
    public class Status : BaseEntity, IStatus
    {
        public char Mood { get; set; }

        public string Text { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }
    }
}

File: Socialite.ReactClient/src/components/Statuses/Status.js
import PropTypes from 'proptypes';

class Status {
    constructor(mood, text, createdAt) {
        this.mood = mood;
        this.text = text;
        this.createdAt = createdAt;
    }
}

Status.propTypes = {
    mood: PropTypes.string,
    text: PropTypes.string,
    createdAt: PropTypes.instanceOf(Date)
};

export default Status;